spec/type_references_and_ownership.md

```diff
@@ -0,0 +1,675 @@
+    
+    // Load the object from etcd
+    obj := scheme.New(gvk)
+    resource := client.Get(req.NamespacedName, obj)
+
+    // Get the owner details
+    armResource := obj.(ARMResource)
+    ownerRef := armResource.Owner()
+
+    // Perform a get from Azure to see current resource state    
+    armId := helpers.GetArmId(resource)
+    objFromAzure := scheme.new(gvk) // We need to provide the empty type to deserialize into
+
+    // Somehow construct a new object of type etcdObject
+    if armTransformer, ok := objFromAzure.(ARMTransformer); ok {
+        result := armTransformer.ToArm("") // This just converts from an empty kube shape to an empty arm shape
+        armClient.GetIt(armId, result)
+    
+        armTransformer.FromArm(ownerRef, result)
+    }
+
+    // Perform a put to update resource state
+    // Walk the owner hierarchy (assuming owner has no owner here for simplicity) to build owner name
+    ownerGvk := ownerRef.ToGvk()
+    owner := scheme.New(ownerGvk)
+    ownerArmResource := owner.(ARMResource)
+    ownerId := owner.Name()
+
+    var toSerialize interface{}
+    toSerialize = resource
+    if armTransformer, ok := toSerialize.(ARMTransformer); ok {
+        toSerialize = armTransformer.ToArm(ownerArmId)
+    }
+    json := json.Marshal(toSerialize)
+    armClient.SendIt(json)
+}
+```
+
+## FAQ
+
+### What happens when a dependent resource specifies an `Owner` that doesn't exist?
+The dependent resource will be stuck in an unprovisioned state with an error stating that the owner doesn't exist.
+If the owner is created, the dependent resource will then be created by the reconciliation loop automatically.
+
+### What happens when a resource contains a link to another resource which doesn't exist?
+The resource with the link will be stuck in an unprovisioned state with an error stating that the linked resource doesn't exist.
+This behavior is the same as for a dependent resource with a non-existant owner.
+
+### How are the CRD entities going to be rendered as ARM deployments?
+There are a few different ways to perform ARM deployments as [discussed in Dependent Resources](#dependent-resources).
+Due to the nature of Kubernetes CRDs, each resource is managed separately and has its own reconcilation loop. It doesn't make sense to try to 
+deploy a single ARM template with the entire resource graph. Each resource will be done in its own deployment (with
+a `dependsOn` specified if required).
+
+### Aren't there going to be races in resource creation?
+Yes. If you have a complex hierarchy of resources (where resources have involved relationships between one another) and submit 
+all of their YAMLs to the operator at the same time it is likely that some requests when sent to ARM will fail because of missing dependencies.
+Those resources that failed to deploy initially will be in an unprovisioned state in Kubernetes, and eventually
+all the resources will be created through multiple iterations of the reconciliation loop. 
```

Comment:
We could also use `DependsOn` in the deployment template to get an ordering that we desire. This might not be needed but it could speed up initial deployment rather than having to wait for the reconciliation loop to fix everything.

Reply:
I think that `DependsOn` requires multiple resources submitted in a single ARM template in order to get that ordering - since our controller instances will each be managing a single type I am pretty sure we don't have an easy path to a unified ARM template with N resources in it?

> Within your template, the dependsOn element enables you to define one resource as a dependent on one or more resources.

https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/define-resource-dependency